src/System.Transactions/src/System/Transactions/TransactionsEtwProvider.cs

```diff
@@ -0,0 +1,877 @@
+        {
+            if (IsEnabled(EventLevel.Informational, ALL_KEYWORDS))
+            {
+                TransactionPromoted(txID.TransactionIdentifier.ToString(), distributedTxID.TransactionIdentifier.ToString());
+            }
+        }
+
+        [Event(TRANSACTION_PROMOTED_EVENTID, Level = EventLevel.Informational, Task = Tasks.Transaction, Opcode = Opcodes.Promoted, Message = "Transaction promoted: transaction ID is {0} and distributed transaction ID is {1}")]
+        private void TransactionPromoted(string txID, string distributedTxID)
+        {
+            WriteEvent(TRANSACTION_PROMOTED_EVENTID, txID, distributedTxID);
+        }
+        #endregion
+
+        #region Transactionstate aborted
+        /// <summary>Trace an event when transaction is aborted.</summary>
+        /// <param name="txID">The transaction ID.</param>
+        [NonEvent]
+        internal void TransactionAborted(TransactionTraceIdentifier txID)
+        {
+            if (IsEnabled(EventLevel.Warning, ALL_KEYWORDS))
+            {
+                TransactionAborted(txID.TransactionIdentifier.ToString());
+            }
+        }
+
+        [Event(TRANSACTION_ABORTED_EVENTID, Level = EventLevel.Warning, Task = Tasks.Transaction, Opcode = Opcodes.Aborted, Message = "Transaction aborted: transaction ID is {0}")]
+        private void TransactionAborted(string txID)
+        {
+            WriteEvent(TRANSACTION_ABORTED_EVENTID, txID);
+        }
+        #endregion
+        public class Opcodes
+        {
+            public const EventOpcode Aborted = (EventOpcode)103;
+            public const EventOpcode Activity = (EventOpcode)101;
+            public const EventOpcode Adjusted = (EventOpcode)118;
+            public const EventOpcode CloneCreate = (EventOpcode)112;
+            public const EventOpcode Commit = (EventOpcode)106;
+            public const EventOpcode Commited = (EventOpcode)104;
+            public const EventOpcode Create = (EventOpcode)100;
+            public const EventOpcode Created = (EventOpcode)119;
+            public const EventOpcode CurentChanged = (EventOpcode)120;
+            public const EventOpcode DependentCloneComplete = (EventOpcode)107;
+            public const EventOpcode Disposed = (EventOpcode)122;
+            public const EventOpcode Done = (EventOpcode)108;
+            public const EventOpcode Enlist = (EventOpcode)127;
+            public const EventOpcode Enter = (EventOpcode)115;
+            public const EventOpcode ExceptionConsumed = (EventOpcode)126;
+            public const EventOpcode Exit = (EventOpcode)116;
+            public const EventOpcode ForceRollback = (EventOpcode)102;
+            public const EventOpcode Incomplete = (EventOpcode)123;
+            public const EventOpcode InDoubt = (EventOpcode)105;
+            public const EventOpcode IntrenalError = (EventOpcode)124;
```

Comment:
Done with the typo. For the ordering, we can do alphabetically and if we need to add more, we just update all eventOpcode after the inserted one.